lib/Sema/CSBindings.cpp

```diff
@@ -25,6 +25,32 @@ ConstraintSystem::determineBestBindings() {
   // Look for potential type variable bindings.
   TypeVariableType *bestTypeVar = nullptr;
   PotentialBindings bestBindings;
+
+  // For purposes of best binding identification
+  // right-hand of `parameter binding` constraint
+  // is equivalent to the left-hand side, which
+  // represents declaration of the parameter and
+  // should drive the search.
+
+  SolverScope scope(*this);
```

Comment:
It's here to rollback changes to the graph and saved bindings at the end of this call.